src/index.ts

```diff
@@ -1,113 +1,59 @@
-import type {
-  AddressLookupArgs,
-  AddressLookupResult,
-  DomainLookupArgs,
-  DomainLookupResult,
-  OnNameLookupHandler,
-} from '@metamask/snaps-sdk';
-import type { AbstractProvider, AddressLike } from 'ethers';
-import { BrowserProvider } from 'ethers';
+import type { OnNameLookupHandler } from '@metamask/snaps-sdk';
+import { parseCaipChainId } from '@metamask/utils';
 
-const ENS_SUPPORTED_CHAINS = ['eip155:1', 'eip155:11155111', 'eip155:17000'];
-const PROTOCOL_NAME = 'Ethereum Name Service';
+import { CAIP_CHAIN_ID_TO_SLIP_44_COIN_TYPE } from './constants';
+import { resolveAddress, resolveDomain } from './resolvers';
+import { configureProvider, isSupportedChain } from './utils';
 
 /**
- * Tries to determine if an address is currently a contract.
- * When unknown it returns `true`.
- * @param provider - AbstractProvider - the provider for the network.
- * @param address - AddressLike - the address to check.
- * @returns Promise<boolean> - true if the given address has bytecode set.
+ * Handler for the `onNameLookup` method.
+ * @param request - The name lookup request.
+ * @returns The name lookup response or null if no resolution is found.
  */
-async function addressIsContract(
-  provider: AbstractProvider,
-  address: AddressLike,
-) {
-  try {
-    const code = await provider.getCode(address, 'pending');
-    return code !== '0x';
-  } catch (error) {
-    console.error(
-      'Unable to determine if resolved ENS address is a contract. Assuming it is and returning nothing.',
-    );
-    return true; // err on the side of caution
-  }
-}
-
-export const onNameLookup: OnNameLookupHandler = async (
-  request: AddressLookupArgs | DomainLookupArgs,
-): Promise<AddressLookupResult | DomainLookupResult | null> => {
+export const onNameLookup: OnNameLookupHandler = async (request) => {
   const { chainId, address, domain } = request;
+  const { namespace, reference } = parseCaipChainId(chainId);
+
+  const chainIsSupported = isSupportedChain(chainId);
 
-  const chainIdInt = parseInt(chainId.split(':')[1] ?? '1', 10);
-  const providerChainId = ENS_SUPPORTED_CHAINS.includes(chainId)
-    ? chainIdInt
-    : 1;
+  const decimalChainId = parseInt(reference, 10);
```

Comment:
If this is a non-EIP reference this will just return `NaN`. Can we rewrite slightly to only parse this if necessary?